src/main/java/org/elasticsearch/transport/local/LocalTransport.java

```diff
@@ -262,27 +262,27 @@ private void handleRequest(StreamInput stream, long requestId, LocalTransport so
         transportServiceAdapter.onRequestReceived(requestId, action);
         final LocalTransportChannel transportChannel = new LocalTransportChannel(this, transportServiceAdapter, sourceTransport, action, requestId, version);
         try {
-            final TransportRequestHandler handler = transportServiceAdapter.handler(action);
-            if (handler == null) {
+            final RequestHandlerReg reg = transportServiceAdapter.getRequestHandler(action);
+            if (reg == null) {
                 throw new ActionNotFoundTransportException("Action [" + action + "] not found");
             }
-            final TransportRequest request = handler.newInstance();
+            final TransportRequest request = reg.newRequest();
```

Comment:
Ultimately I think we should have `Streamable` return a new instance from readFrom then we can have everything immutable and get factories for free